### ISSUE TITLE
seldom crash

### ISSUE THREAD

**Issue by alparp27** [DataDog/dd-sdk-android#234]
Thanks for taking the time for reporting an issue!

**Describe what happened**
crash	
 
```
Session summary
 3.1.2.2 (1603311495)
 7.0
 LG Stylo 3 Plus
 Apr 21, 2020, 11:57:00 PM
Stack trace


Fatal Exception: java.util.ConcurrentModificationException
java.util.LinkedHashMap$LinkedHashIterator.nextEntry (LinkedHashMap.java:411)
java.util.LinkedHashSet.<init> (LinkedHashSet.java:169)
kotlin.collections.CollectionsKt___CollectionsKt.toMutableSet (_Collections.kt:1452)
kotlin.collections.CollectionsKt___CollectionsKt.union (_Collections.kt:1467)
com.datadog.android.core.internal.data.file.FileReader.readNextFile (FileReader.kt:72)
com.datadog.android.core.internal.data.file.FileReader.readNextBatch (FileReader.kt:31)
com.datadog.android.core.internal.data.upload.DataUploadRunnable.run (DataUploadRunnable.kt:36)
android.os.Handler.handleCallback (Handler.java:751)
android.os.HandlerThread.run (HandlerThread.java:61)
```
**Steps to reproduce the issue:**

**Describe what you expected:**

**Additional context**
 - Android OS version:
 - Device Model:
 - Datadog SDK version:
 - Versions of any other relevant dependencies (OkHttp, …): 
 - Proguard configuration: 
 - Gradle Plugins:

**Comment by xgouchet** [comment]
Hey @alparp27 , thanks for submitting this issue, we'll look into it very soon. 
Do you have anymore information on this issue (scenario, context, …) ?

**Comment by mariusc83** [comment]
Hey @alparp27 we figured it out where the problems might come from and we will let you know once a hotfix release will be available. Thank you again for flagging this !!

**Comment by mariusc83** [comment]
@alparp27 we will need one extra bit of information from you, can you please let us know the Datadog SDK version number for which you experienced the crash ? Thanks !!

**Comment by alparp27** [comment]
1.3.0

**Comment by mariusc83** [comment]
Thank you for all your effort @alparp27. The issue should be solved now in version 1.3.1.
